src/storage-service.ts

```diff
@@ -47,6 +51,12 @@ export class MockStorageService implements StorageService {
     console.log('[MockSynapse] CommP:', commpString)
     console.log('[MockSynapse] Download options:', options)
 
+    if (options?.withCDN !== false && (this._withCDN || options?.withCDN)) {
```

Comment:
It's a bit awkward to both support this option in the synapse constructor and the upload arguments. This makes it so if either is true, the CDN is used, unless a `false` was passed to this call.

Reply:
Tests need to be added for this condition. I want to get approval on the direction first before adding more code

Reply:
We don't have to have the option in both places, we could just do it one or the other? It could even show upas an option to `upload()` and `download()`, and that might end up being too many places and too confusing. I'm fine with keeping the number of these down. What do you think?

Reply:
🤷‍♂️ I think from the user's perspective, what you have intuitively make sense: You can define options on the instance, but overwrite on demand in the specialized function.